javascript/javascript1/week3/myFavoriteSongs.js

```diff
@@ -0,0 +1,62 @@
+const songDatabase = [{
+    songId: 1,
+    title: 'My baby',
+    artist: 'Soggy socks',
+  },
+  {
+    songId: 2,
+    title: '3 nails in wood',
+    artist: 'The carpenters',
+  },
+  {
+    songId: 3,
+    title: 'Blacker than black',
+    artist: 'Instant coffee',
+  },
+  {
+    songId: 4,
+    title: 'When is enough too little?',
+    artist: 'The spies girls',
+  },
+];
+const myPlaylist = [];
+
+function myList(index)
+    {
+        const newSong=songDatabase[index];
+        myPlaylist.push(newSong);
+    }
+console.log("My List:");
+
+function addSongToDatabase(song) 
+{
+    songDatabase.push(song);
+}
+
+  function getSongByTitle(title)
+   {
+    index = songDatabase.findIndex(x => x.title === title);
+    const newSearchedSong = songDatabase[index];
+    return newSearchedSong;
+  }
+  
+  const searchedSong = getSongByTitle("3 nails in wood");
+  console.log(searchedSong);
+  
+  const searchedSong2 = getSongByTitle("My baby");
+  console.log(searchedSong2);
+
+  
+  
+  
+  function addSongToMyPlaylist(title)
+   {
+    index = songDatabase.findIndex(x => x.title === title);
```

Comment:
index could be -1 (not found) => you need to do a condition for that.

Reply:
I changed the code here...
https://github.com/Aswini-D3/hyf-homework/blob/javascript/javascript1/week3/javascript/javascript1/week3/myFavoriteSongs.js
Please check...

Reply:
Not sure what you're doing there. the for loop is OK, but your original findIndex solution was better. All you need is to add a check for -1 :) Please redo with `findIndex` or `indexOf` without the for loop.